src/bun.js/test/jest.zig

```diff
@@ -1172,6 +1183,29 @@ pub const DescribeScope = struct {
         return .js_undefined;
     }
 
+    fn markChildrenHaveTests(this: *DescribeScope) void {
+        var parent: ?*DescribeScope = this;
+        while (parent) |scope| {
+            if (scope.children_have_tests) break;
+            scope.children_have_tests = true;
+            parent = scope.parent;
+        }
+    }
+
+    // TODO: combine with shouldEvaluateScope() once we make beforeAll run with the first scheduled test in the scope.
+    fn shouldRunBeforeAllAndAfterAll(this: *const DescribeScope) bool {
+        if (this.children_have_tests) {
+            return true;
+        }
+
+        if (Jest.runner.?.hasTestFilter()) {
+            // All tests in this scope were filtered out.
+            return false;
+        }
+
+        return true;
```

Comment:
This seems fine for Bun, but here Jest throws an error "beforeAll() may not be used in a describe block containing no tests."